packages/block-editor/src/components/inserter/search-results.native.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * External dependencies
+ */
+import {
+	FlatList,
+	View,
+	TouchableHighlight,
+	TouchableWithoutFeedback,
+	Dimensions,
+} from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+import {
+	BottomSheet,
+	BottomSheetConsumer,
+	InserterButton,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style.scss';
+
+const MIN_COL_NUM = 3;
+
+function InserterSearchResults( { items, onSelect, onClose } ) {
+	const [ numberOfColumns, setNumberOfColumns ] = useState( MIN_COL_NUM );
+	const [ itemWidth, setItemWidth ] = useState();
+	const [ maxWidth, setMaxWidth ] = useState();
+
+	useEffect( () => {
+		Dimensions.addEventListener( 'change', onLayout );
+		return () => {
+			Dimensions.removeEventListener( 'change', onLayout );
+		};
+	}, [] );
+
+	function calculateItemWidth() {
+		const {
+			paddingLeft: itemPaddingLeft,
+			paddingRight: itemPaddingRight,
+		} = InserterButton.Styles.modalItem;
+		const { width } = InserterButton.Styles.modalIconWrapper;
+		return width + itemPaddingLeft + itemPaddingRight;
```

Comment:
What are your thoughts about using nested function (like we have here) as opposed to unexported top-level functions? Just asking to learn more, not because I think this needs to change. My personal preference is to favor top-level functions, but that is more a general code preference because I like how that makes things a bit painful if your function gets overly complicated (i.e., you have to start passing a lot of arguments to the function). Perhaps there are JS-specific reasons to prefer nested functions?

Reply:
I believe it's become a common pattern i JS due to scope. I find it nice for currying functions that get passed around in React.